x/mongo/driver/operation.go

```diff
@@ -355,6 +357,11 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 			}
 		}
 	}
+	// If context is a Timeout context, automatically set retries to -1 (infinite) unless RetryMode
+	// is set to RetryNever.
+	if internal.IsTimeoutContext(ctx) && (op.RetryMode == nil || *op.RetryMode != RetryNever) {
```

Comment:
Is there a reason we can't use `RetryNone` here? It could be confusing to have a `RetryNone` retry mode that is not applicable when the context has a timeout. Perhaps if we do need to differentiate between these two things (i.e. `RetryNever` and `RetryNone`) it may makes more sense to do something like 

- `RetryNone` - disables retrying
- `RetryNoneExceptTimeout` (or w/e, just less strict RetryNone) - disables retrying if Timeout is not specified 

That is, we keep `RetryNone` as the most strict "we do not retry" option and add the slightly more relaxed `RetryNonExcepTimeout` type. Then we can make updates like 

```go
if internal.IsTimeoutContext(ctx) && (op.RetryMode == nil || *op.RetryMode != RetryNone) {
// ..
}
```
and 

```go
if *op.RetryMode > RetryNoneExceptTimeout {
// ...
}
```
What are your thoughts?